client/state/data-layer/wpcom/activity-log/delete-credentials/index.js

```diff
@@ -26,14 +31,25 @@ export const request = ( { dispatch }, action ) => {
 	);
 };
 
-export const success = ( { dispatch }, action ) => {
+export const success = ( { dispatch }, { siteId }, { rewind_state } ) => {
 	dispatch( {
 		type: JETPACK_CREDENTIALS_STORE,
 		credentials: {
 			main: null,
 		},
-		siteId: action.siteId,
+		siteId,
 	} );
+
+	// the API transform could fail and the rewind data might
```

Comment:
This section is repeated 3 times. I assume that this is not extracted because this is temporary?

Reply:
it's duplicated because it's a tricky call I wasn't sure how to make. this _is_ temporary in one sense because we just bolted-on the `/rewind` API response to these existing ones but on the other hand our goal is to have them eventually _just_ return the `/rewind` response in its entirety.

that means we'll probably end up keeping this here. now, if the entire response turns into the `/rewind` response then we'll just pass `transformApi` along as the `fromApi` function and things will be simpler, but right now it's not guaranteed to exist so we have to manually check if it's there.

making it further difficult is that the `fromApi` family of functions are designed to `throw` if they have problems parsing and transforming their input, so we need the `try/catch` here just in case.

now, I _could_ have extracted this into a separate utility function but I felt at the time of writing that doing so might actually obscure what's happening and that by repeating the code it would be much more obvious what we're doing and why.

I'd be very open to hear your thoughts on this!

(also, thanks a million for reviewing! 👏 )